service/src/main/java/com/fecklessweasel/service/objectmodel/FileMetadata.java

```diff
@@ -0,0 +1,281 @@
+     this.fid = fid;
+     this.user = user;
+     this.course = course;
+     this.university = university;
+     this.creationDate = creationDate;
+     this.rating = rating;
+   }
+    /**
+     * inserts the filemetadata as a new entry into the filemetadata table, returns the object
+     * wrapper that represents the filemetadata.
+     */
+   public static FileMetadata create(Connection sql,
+                        String user,
+                        String course,
+                        String university,
+                        Date creationDate) throws ServiceException{
+    OMUtil.sqlCheck(sql);
+    OMUtil.nullCheck(user);
+    OMUtil.nullCheck(course);
+    OMUtil.nullCheck(university);
+    OMUtil.nullCheck(creationDate);
+
+    //TODO Add validation for data members
+
+    int fid = FileMetadataTable.insertFileData(sql, user, course, university, creationDate, INITIAL_RATING);
+    return new FileMetadata(fid, user, course, university, creationDate, INITIAL_RATING);
+  }
+  /**
+   * Look up a specific file, and return an object wrapper around its metadata row
+   *@param sql The Sql connection to the FecklessWeaselDB
+   *@param fid The Files unique identifier
+   *@return The FileMetadata object representation of the row found
+   */
+  public static FileMetadata lookup(Connection sql, int fid) throws ServiceException{
+    OMUtil.sqlCheck(sql);
+    ResultSet result = FileMetadataTable.lookUpFile(sql, fid);
+
+    try{
+        //if no tuples returned, throw error.
+        if (!result.next()) { throw new ServiceException(ServiceStatus.APP_FILE_NOT_EXIST); }
+
+        FileMetadata fileData = new FileMetadata(result.getInt("fid"),
+                                                 result.getString("user"),
+                                                 result.getString("course"),
+                                                 result.getString("university"),
+                                                 result.getDate("creation_date"),
+                                                 result.getInt("rating"));
+        result.close();
+        return fileData;
+    } catch (SQLException ex) {
+      throw new ServiceException(ServiceStatus.DATABASE_ERROR, ex);
+    }
+  }
+
+  /**
+   * Look up all files in a specified course
+   *@param sql The Sql connection to the FecklessWeaselDB
+   *@param course The course whose file info we want
+   *@param university The University that has the course
+   *@return A List of FileMetadata objects that represent each files info
+   */
+  public static List<FileMetadata> lookUpCourseFiles(Connection sql, String course, String university)
+    throws ServiceException{
+    OMUtil.sqlCheck(sql);
+    OMUtil.nullCheck(course);
+    OMUtil.nullCheck(university);
+
+    ResultSet results = FileMetadataTable.lookUpCourseFiles(sql, course, university);
+    ArrayList<FileMetadata> listOfFiles = new ArrayList<FileMetadata>();
+
+    try{
+      while( results.next() ){
```

Comment:
nit: space between while and "(" and no spaces between "(" and ")" and stuff inside.